### PR TITLE
Audit fixing

### PR DESCRIPTION
**Pull request by catalinnnn**
Fix update last_reward_block_nonce at each step
Added check for fee configuration in pair sc
Added possibility for user to check his temporary funds in dex proxy sc

### CODE REVIEW

File: distribution/sc-proxy-dex/src/proxy_pair.rs [hunk 1]
```diff
@@ -561,13 +586,27 @@ pub trait ProxyPairModule: proxy_common::ProxyCommonModule {
...
+        &self,
+        address: &Address,
+    ) -> MultiResultVec<GenericEsdtAmountPair<Self::BigUint>> {
+        let vec: Vec<GenericEsdtAmountPair<Self::BigUint>> = self
+            .temporary_funds(address)
+            .iter()
+            .map(|x| GenericEsdtAmountPair {
+                token_id: x.0 .0,
+                token_nonce: x.0 .1,
+                amount: x.1,
```
Comment (andrei-marinica) [inline distribution/sc-proxy-dex/src/proxy_pair.rs:599]: It's okay as it is but could be made a little easier to read like this
```suggestion
                let (key, amount) = x;
                let (token_id, token_nonce) = key;
                GenericEsdtAmountPair { token_id, token_nonce, amount }
```

**Review by andrei-marinica — APPROVED** [review]

File: distribution/sc-proxy-dex/src/proxy_pair.rs (hunk 1 above)
Comment (catalinnnn) [reply distribution/sc-proxy-dex/src/proxy_pair.rs]: Didn't know about this syntax. Thank you. Fixed

File: distribution/sc-proxy-dex/src/proxy_pair.rs [hunk 2]
```diff
@@ -561,13 +586,31 @@ pub trait ProxyPairModule: proxy_common::ProxyCommonModule {
...
+            .map(|x| {
+                let (key, amount) = x;
+                let (token_id, token_nonce) = key;
+                GenericEsdtAmountPair {
+                    token_id,
+                    token_nonce,
+                    amount,
+                }
+            })
+            .collect();
```
Comment (claudiu725) [inline distribution/sc-proxy-dex/src/proxy_pair.rs:605]: You can `.collect()` directly into a `MultiResultVec`, without needing to use an intermediate `Vec`

**Review by claudiu725 — CHANGES_REQUESTED** [review]

File: distribution/sc-proxy-dex/src/proxy_pair.rs (hunk 2 above)
Comment (catalinnnn) [reply distribution/sc-proxy-dex/src/proxy_pair.rs]: Done

**Review by sasurobert — APPROVED** [review]

**Pull request merged**